platform/security/idp/security-idp-plugin/security-idp-plugin-subjectconfirmation/src/main/java/org/codice/ddf/security/idp/plugin/subjectconfirmation/SubjectConfirmationPresignPlugin.java

```diff
@@ -88,27 +87,13 @@ private Audience buildAudience(AudienceBuilder audienceBuilder, String uri) {
     return audience;
   }
 
-  private void addAudiences(
-      AuthnRequest authnRequest,
-      Response response,
-      List<String> spMetadata,
-      Set<Binding> supportedBindings) {
+  private void addAudiences(AuthnRequest authnRequest, Response response) {
     AudienceBuilder audienceBuilder = new AudienceBuilder();
     AudienceRestrictionBuilder audienceRestrictionBuilder = new AudienceRestrictionBuilder();
 
-    Collection<EntityInformation> entityInformationCollection =
-        parseServiceProviderMetadata(spMetadata, supportedBindings).values();
-
-    List<Audience> audienceList =
-        entityInformationCollection
-            .stream()
-            .map(
-                ei ->
-                    ei.getAssertionConsumerService(
-                        authnRequest, null, authnRequest.getAssertionConsumerServiceIndex()))
-            .map(ServiceInfo::getUrl)
-            .map(uri -> buildAudience(audienceBuilder, uri))
-            .collect(Collectors.toList());
+    // According to the SAML spec, on an AuthnRequest, "[t]he <Issuer> element MUST be present and
+    // MUST contain the unique identifier of the requesting service provider".
+    Audience audience = buildAudience(audienceBuilder, authnRequest.getIssuer().getValue());
```

Comment:
❗️ Shouldn't we have a unit test for the AudienceRestrictions? I think it's especially important here to make sure we stay spec compliant.

Reply:
I'd suggest we kick that slightly down the road and make it part of the definition of done on DDF-3494.

Reply:
I agree since a unit test here will just add more tech debt that needs to be reworked 👍